datasets/trigger_response.py

```diff
@@ -75,21 +85,40 @@ def generate_examples(self, num_examples):
 
         return examples
 
+    def evaluate_single(self, actual: str, expected: str) -> tuple[int, str]:
+        if expected in actual or rouge_score(expected, actual) > 0.5:
+            return 1, f"'{expected}' is in the response."
+        context = [make_user_message(eval_prompt.format(message=actual, sentence=expected))]
+        eval_str = self.ask_llm(context)
+        try:
+            eval_json = sanitize_and_parse_json(eval_str)
+            present = eval_json["present"]
+            if present:
+                present = rouge_score(expected, eval_json["sentence"]) > 0.75
+        except (ValueError, JSONDecodeError, KeyError) as exc:
+            return 0, f"Could not evaluate due to a JSON parsing error: {repr(exc)}"
+        not_str = "" if present else "not "
+        return int(present), f"'{expected}' is {not_str}in the response."
+
```

Comment:
Given the conversation for `TriggerResponse`, can we just string match here?

Reply:
I almost say "yes", but then I remembered about our conversation last week, about us constantly switching back and forth between string matching and GPT evaluations 😛 The thing is, agents sometimes stick the user's name in the middle of the sentence, keeping the rest verbatim. You could get around that with rouge-L scores, but sentence lengths affect it quite a bit, and agents usually say more than the sentence alone. So that's why the combination of a loose and somewhat flexible sequence extraction (GPT) + the high rouge score on that sentence.